rfcs/0031-ppc64-platforms.md

```diff
@@ -0,0 +1,86 @@
+---
+feature: ppc64-platforms
+start-date: 2018-08-23
+author: CrystalGamma
+co-authors: (find a buddy later to help out with the RFC)
+related-issues: https://github.com/NixOS/nixpkgs/pull/45340 https://github.com/NixOS/nixpkgs/pull/45472 https://github.com/NixOS/nixpkgs/pull/45474 https://github.com/NixOS/nixpkgs/pull/45475
+---
+
+# Summary
+[summary]: #summary
+
+Bring support for ppc64 platforms, initially especially ppc64le, to nixpkgs/NixOS.
+
+# Motivation
+[motivation]: #motivation
+
+NixOS/nixpkgs has long prided itself to be a general-purpose OS/package repository, allowing its packages to be used across a broad range of systems, and providing more packages than ever.
+
+With alternative architectures such as OpenPOWER and RISC-V gaining traction, now is the time to add support for those architectures.
+Specifically motivating this request is the OpenPOWER, or powerpc64le, architecture—and its embodiment in the POWER series of microprocessors from IBM.
+Fully open source commercially available systems have recently become widely available for this platform and there are active Nix users who desire support.
+
+# Detailed design
+[design]: #detailed-design
+
+The purpose of this RFC is to create consensus for general acceptance of ppc64(le) related pull requests (as long as they are technically reasonable).
+A general commitment for contributions to not break ppc64(le) is not necessary as long as the architecture stays in only niche use.
+This especially means that building a Continuous Integration infrastructure for ppc64le is not required in the near future.
+
+Bringing support for ppc64(le) platforms will be done as follows:
+
+The first step for implementing support is extending lib/ to recognize powerpc64(le) as a CPU architecture and allowing bootstrap files for the architecture to be (cross-)built and a stdenv based on those bootstrap files to be built.
+This has already been merged in [nixpkgs#45340](https://github.com/NixOS/nixpkgs/pull/45340).
```

Comment:
The more architectures we support, the harder it's going to be to upgrade gcc and other heavily-patched packages. Maintainers don't have access to the more esoteric platforms.

How do you propose that patches be handled if for example gcc has to be upgraded quickly for security-related reasons?

Reply:
Maybe lib/systems/platforms.nix should be amended with an extra "officiallySupported" flag.